src/gdf_table.cuh

```diff
@@ -41,105 +43,297 @@ public:
+            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
+            my_elem = other_elem;
             break;
           }
         case GDF_INT64:
           {
-            sprintf(row,"%ld", static_cast<int64_t*>(d_columns_data[i])[row_index]);
+            using col_type = int64_t;
+            col_type & my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
+            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
+            my_elem = other_elem;
             break;
           }
         case GDF_FLOAT32:
           {
-            sprintf(row,"%f", static_cast<float*>(d_columns_data[i])[row_index]);
+            using col_type = float;
+            col_type & my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
+            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
+            my_elem = other_elem;
             break;
           }
         case GDF_FLOAT64:
           {
-            sprintf(row,"%f", static_cast<double*>(d_columns_data[i])[row_index]);
+            using col_type = double;
+            col_type & my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
+            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
+            my_elem = other_elem;
             break;
           }
         case GDF_DATE32:
           {
-            sprintf(row,"%d", static_cast<int32_t*>(d_columns_data[i])[row_index]);
+            using col_type = int32_t;
+            col_type & my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
+            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
+            my_elem = other_elem;
             break;
           }
         case GDF_DATE64:
           {
-            sprintf(row,"%ld", static_cast<int64_t*>(d_columns_data[i])[row_index]);
+            using col_type = int64_t;
+            col_type & my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
+            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
+            my_elem = other_elem;
             break;
           }
         case GDF_TIMESTAMP:
           {
-            sprintf(row,"%ld", static_cast<int64_t*>(d_columns_data[i])[row_index]);
+            using col_type = int64_t;
+            col_type & my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
+            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
+            my_elem = other_elem;
             break;
           }
         default:
-          assert(false && "Attempted to compare unsupported GDF datatype");
+          printf("Attempted to copy column of unsupported GDF datatype\n");
```

Comment:
Same, should this be an exception, or at least printed to stderr?

Reply:
Unfortunately since this is a device function, printing to stderr or throwing an exception isn't possible. Instead I've just made all the device functions return gdf_error when possible. We should spend some time thinking about how we want to do error reporting from inside kernels.

Reply:
Uggh yeah, definitely.  I would think that if you can get to stdout, then stderr would also be available, but I suppose not.